code/ATMOSPHERICS/gas_types.dm

```diff
@@ -0,0 +1,81 @@
+var/list/hardcoded_gases = list("o2","n2","co2","plasma") //the main four gases, which were at one time hardcoded
+
+/proc/meta_gas_list()
+	var/meta_list = new /list
+	for(var/gas_path in subtypesof(/datum/gas))
+		var/list/gas_info = new(4)
+		var/datum/gas/g = gas_path
+
+		gas_info[1] = initial(g.specific_heat)
+		gas_info[2] = initial(g.name)
+		gas_info[3] = initial(g.moles_visible)
+		if(gas_info[3] != null)
+			gas_info[4] = new /obj/effect/overlay/gas(initial(g.gas_overlay))
```

Comment:
If I do that how am I supposed to change the icon state of carbon dioxide to a sprite saying "WTF?"